hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/events/Event.java

```diff
@@ -20,16 +20,16 @@
 /**
  * Identifier of an async event.
  *
- * @param <PAYLOAD> THe message payload type of this event.
+ * @param <P> THe message payload type of this event.
  */
-public interface Event<PAYLOAD> {
+public interface Event<P> {
```

Comment:
Can we leave this generic type as PAYLOAD, instead of a single character (even though sonar reports it), it looks good while reading code.

Reply:
I was kind of afraid of making this part of the change. ;)  I think the rationale is that ALL_CAPS are for constants.  I personally prefer the compliant version.  However, I would be happy to hear from more people about this.  I will change it back to the original (and mark the Sonar issues as "won't fix") if others also prefer the long version.  Is that OK?

Reply:
I like the more meaningful names, it's easier to understand what is the parameter. I think sonar cloud rules can be adjusted by admin privileges...

Reply:
+1 for meaningful naming.

Reply:
Have you guys read the reasoning for this Rule. Sonar has rationale for each rule. From the rule book.

"Shared naming conventions make it possible for a team to collaborate efficiently. Following the established convention of single-letter type parameter names helps users and maintainers of your code quickly see the difference between a type parameter and a poorly named class.

This rule check that all type parameter names match a provided regular expression. The following code snippets use the default regular expression."

It is basically saying that it wants a regular expression that matches a Type parameter, so people and tools get consistent experience. So if your rule is some like I need more chars but capital letter, then we need to spell out that rule. 

From my own perspective; since I am very lazy, if someone else has a written a rule for me, it is far easier for me to follow; rather than think, debated and ultimately form a new rule. 

So either one of you need to define the new rule -- for the TYPES in Java or let this be. I am open with either approach.

Reply:
Here is one guy who actually agrees with you .. 
https://www.vojtechruzicka.com/break-java-generics-naming-convention/

But remember, this is the standard Sun/Oracle Java convention. While I am no Java coding conventions fan; just reminding that inventing a new rule and getting it consensus on something like this is generally painful. But feel free to be my guest.

Reply:
Thanks @anuengineer for looking up the rule and researching it on the net, too.  Since there was no agreement on changing the names, I updated to the previous state as promised.

> Here is one guy who actually agrees with you ..
> https://www.vojtechruzicka.com/break-java-generics-naming-convention/

I think it's worth noting that none of his three options is "use ALL_CAPS" (only mentioned in the comments).